portal/test/[locale]/staking-dashboard/_utils/aprCalculations.test.ts

```diff
@@ -0,0 +1,359 @@
+      expect(result[i]).toBeGreaterThan(BigInt(0))
+    }
+
+    // Should decay gradually but never reach zero in the 61 epochs
+    expect(result[0]).toBeGreaterThan(result[30])
+    expect(result[30]).toBeGreaterThan(result[60])
+    expect(result[60]).toBeGreaterThan(BigInt(0))
+  })
+
+  it('should handle position already unlocked (all zeros)', function () {
+    const result = calculateRewardWeightDecay({
+      currentBalance: BigInt('100000000000000000000'),
+      currentTimestamp: BigInt(1731772800),
+      lockEndTimestamp: BigInt(1731772800 - 1000), // Already unlocked
+    })
+
+    // All epochs should be zero
+    for (let i = 0; i < 61; i++) {
+      expect(result[i]).toBe(BigInt(0))
+    }
+  })
+
+  it('should handle very small balance without precision loss', function () {
+    const smallBalance = BigInt('1000') // Very small amount
+
+    const result = calculateRewardWeightDecay({
+      currentBalance: smallBalance,
+      currentTimestamp: BigInt(1731772800),
+      lockEndTimestamp: BigInt(1731772800 + 30 * secondsPerEpoch),
+    })
+
+    // Should still calculate values (not all zeros due to rounding)
+    expect(result[0]).toBe(smallBalance)
+    expect(result[15]).toBeGreaterThan(BigInt(0))
+  })
+
+  it('should handle position unlocking exactly at epoch boundary', function () {
+    const epochsUntilUnlock = 10
+
+    const result = calculateRewardWeightDecay({
+      currentBalance: BigInt('100000000000000000000'),
+      currentTimestamp: BigInt(1731772800),
+      lockEndTimestamp: BigInt(
+        1731772800 + epochsUntilUnlock * secondsPerEpoch,
+      ), // Exactly 10 epochs
+    })
+
+    // Epoch 9 should have value, epoch 10 should be zero
+    expect(result[9]).toBeGreaterThan(BigInt(0))
+    expect(result[10]).toBe(BigInt(0))
+  })
+})
+
+describe('calculateApr', function () {
+  const DAILY_REWARDS = 8333
+
+  const totalSupplyPerEpoch = [
+    11322220, 11115402, 10908662, 10702202, 10497204, 10293065, 10088943,
+    9884831, 9680801, 9480645, 9282703, 9084989, 8887363, 8689749, 8492143,
+    8294577, 8097046, 7899532, 7699575, 7708176, 7652489, 7598632, 7546765,
+    7497867, 7451352, 7404896, 7359364, 7315207, 7271102, 7227094, 7183332,
+    7139786, 7096240, 7052695, 7009150, 6965606, 6922063, 6878520, 6834978,
+    6791437, 6747898, 6704359, 6660819, 6617280, 6573741, 6530202, 6486663,
+    6443126, 6399591, 6356066, 6312541, 6269020, 6225499, 6181977, 6138456,
+    6094935, 6051415, 6007894, 5964374, 5920852, 5877330,
+  ]
+
+  // Convert to rewards per veHEMI (dailyRewards / totalSupply)
+  const MOCK_REWARDS_PER_VEHEMI = totalSupplyPerEpoch.map(
+    supply => DAILY_REWARDS / supply,
+  )
+
+  it('should calculate APR correctly for position unlocking after 1 year', function () {
+    const currentBalance = BigInt('100000000000000000000') // 100 veHEMI
+    const lockedAmount = BigInt('100000000000000000000') // 100 HEMI
+    const epochsUntilUnlock = 100 // Unlocks after 61 epochs
```

Comment:
Just checking: Is this comment correct? The variable seems to imply that it unlocks after 100 epochs, but the comment says "unlocks after 61"